horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/user/DatabaseUserBackend.java

```diff
@@ -27,46 +25,37 @@
 import java.util.Optional;
 import java.util.function.Function;
 
+import static io.hyperfoil.tools.horreum.svc.UserServiceImpl.getTeamPrefix;
 import static java.util.stream.Collectors.toList;
 import static java.util.stream.Collectors.toSet;
 
-@PermitAll
 @ApplicationScoped
-@Unremovable
-@IfBuildProperty(name = "horreum.roles.provider", stringValue = "database")
-public class DatabaseUserServiceImpl implements UserService {
+@LookupIfProperty(name = "horreum.roles.provider", stringValue = "database")
+public class DatabaseUserBackend implements UserBackEnd {
 
-   private static final Logger LOG = Logger.getLogger(DatabaseUserServiceImpl.class);
+   @Inject
+   SecurityIdentity identity;
 
-   @Inject SecurityIdentity identity;
+   private static final Logger LOG = Logger.getLogger(DatabaseUserBackend.class);
 
-   private static UserData toUserInfo(UserInfo info) {
-      return new UserData("", info.username, info.fistName, info.lastName, info.email);
+   private static UserService.UserData toUserInfo(UserInfo info) {
+      return new UserService.UserData("", info.username, info.fistName, info.lastName, info.email);
    }
 
-   @Authenticated
-   @Override public List<String> getRoles() {
-      return identity.getRoles().stream().toList();
-   }
-
-   @Authenticated
-   @WithRoles(extras = Roles.HORREUM_SYSTEM)
```

Comment:
need to keep `@WithRoles` annotation because of row level security

Reply:
Reinstated @WithRoles